packages/trader/src/AppV2/Hooks/useActiveSymbols.ts

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect, useCallback, useRef } from 'react';
+import { WS, pickDefaultSymbol } from '@deriv/shared';
+import { useStore } from '@deriv/stores';
+import { ActiveSymbols } from '@deriv/api-types';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { usePrevious } from '@deriv/components';
+
+type TUseActiveSymbols = {
+    contract_type?: string[];
+    barrier_category?: string[];
+};
+//TODO: contract_type and barrier_category need to come from trade-store after calling contracts_for
+const useActiveSymbols = ({ contract_type = [], barrier_category = [] }: TUseActiveSymbols) => {
+    const [activeSymbols, setActiveSymbols] = useState<ActiveSymbols | []>([]);
+    const { client, modules } = useStore();
+    const { is_logged_in } = client;
+
+    const trader = useTraderStore();
+
+    const default_symbol_ref = useRef<string>('');
```

Comment:
nit-pick:
```suggestion
    const default_symbol_ref = useRef('');
```
the type will be inferred :)